### ISSUE TITLE
Featured Image attributes not applied for `useFirstImageFromPost`

### ISSUE THREAD

**Issue by ajaydsouza** [WordPress/gutenberg#63454]
### Description

If the `useFirstImageFromPost` attribute is set to true for the Post Featured Image, then the style attributes like width, height and object-fit are not applied to the first image. These get applied to the featured image if it set.

### Step-by-step reproduction instructions

1. Insert the Query Loop block with the featured image. Ideally have a few posts which have featured images and others which don't have featured images, but do have an image in the post content.
2. Select the Featured Image to modify its settings.
3. Edit the image parameters in the sidebar, e.g. setting width, height, object-fit etc.
4. Enter the Code Editor and edit the featured image markup to set useFirstImageFromPost to true:
e.g. `<!-- wp:post-featured-image {"isLink":true,"aspectRatio":"1","width":"","height":"","useFirstImageFromPost":true} /-->`
5. Save the post and view it in the frontend. The images that have a featuerd image have the attributes set. The ones without it, don't have the attribute set.

### Environment info

WordPress 6.6-RC3 and 6.5.5

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by ndiego** [comment]
Yes, I can confirm. First post has a featured image set. The second does not but has an image on the post content.

| Editor | Frontend |
|-|-|
|<img width="868" alt="image" src="https://github.com/user-attachments/assets/f33f1c1c-5d6c-4e17-9aa5-541dc6c92172">|<img width="906" alt="image" src="https://github.com/user-attachments/assets/cea4b2a7-2d25-4aac-b764-fd128bbbbbed">|

**Comment by annezazu** [comment]
Noting this PR underway to address this: https://github.com/WordPress/gutenberg/pull/64900
